lib/widgets/action_sidebar.dart

```diff
@@ -0,0 +1,219 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class ActionSideBar extends StatefulWidget {
+  const ActionSideBar(this.actionButtons);
+
+  final List<Widget> actionButtons;
+
+  @override
+  _ActionSideBarState createState() => _ActionSideBarState();
+}
+
+class _ActionSideBarState extends State<ActionSideBar>
+    with SingleTickerProviderStateMixin {
+  _ActionSideBarState({this.isExtended = false});
+
+  bool isExtended;
+
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 250),
+      vsync: this,
+    );
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void onTap() {
+    setState(() {
+      isExtended = !isExtended;
+
+      if (isExtended) {
+        _controller.forward();
+      } else {
+        _controller.reverse();
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (MediaQuery.of(context).size.width > 1100) {
+      isExtended = true;
+    }
+
+    return AnimatedPositioned(
+      duration: const Duration(milliseconds: 250),
+      left: isExtended ? 0 : 50,
+      right: isExtended ? 0 : -50,
+      top: 0,
+      bottom: 0,
+      child: ConstrainedBox(
+        constraints: const BoxConstraints(maxWidth: 100),
+        child: Stack(
+          children: [
+            Positioned(
+              right: 0,
+              top: 0,
+              bottom: 0,
+              child: ConstrainedBox(
+                constraints: const BoxConstraints(maxWidth: 50),
+                child: Container(
+                  child: Column(
+                    children: widget.actionButtons,
+                  ),
+                  decoration: BoxDecoration(
+                    color: Theme.of(context).bottomAppBarColor.withOpacity(0.9),
+                    boxShadow: [
+                      BoxShadow(
+                        color: Colors.black.withOpacity(0.25),
+                        spreadRadius: 3,
+                        blurRadius: 3,
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+            if (MediaQuery.of(context).size.width < 1100)
+              Positioned(
+                right: 50,
+                child: ActionBarButton(onTap, _controller),
+              )
+            else
+              const SizedBox.shrink(),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class ActionBarButton extends StatelessWidget {
+  const ActionBarButton(this.onTap, this.controller,
+      {this.maxWidth = 50, this.maxHeight = 50});
```

Comment:
Again defining parameters that are never used? Or am I missing something?

Reply:
Removed values from constructor:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/8c0e90bd1db807c70ea5422527345d766711da4c#diff-91af2d9745e86244940959ec39c99cca9b8b43c1502042d85f16fcc6a311030bR135-R137